test_util/helpers.py

```diff
@@ -189,3 +248,15 @@ def remove_file():
     # Attempt to remove the file upon normal interpreter exit.
     atexit.register(remove_file)
     return temp_path
+
+
+def lazy_property(property_fn):
```

Comment:
Can  `functools.lru_cache`  be used here? 

https://docs.python.org/3/library/functools.html#functools.lru_cache

Or is `_lazy_prop` completely different?